src/GitHub.InlineReviews/ViewModels/GlyphMarginViewModel.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.ComponentModel;
+using Microsoft.VisualStudio.Text.Editor;
+using GitHub.Settings;
+using GitHub.Services;
+using GitHub.InlineReviews.Tags;
+
+namespace GitHub.InlineReviews.ViewModels
+{
+    public sealed class GlyphMarginViewModel : IGlyphMarginViewModel, IDisposable
+    {
+        readonly IPackageSettings packageSettings;
+        readonly IPullRequestSessionManager sessionManager;
+        readonly IWpfTextViewHost wpfTextViewHost;
+        readonly bool isDiffView;
+
+        bool visible;
+
+        public GlyphMarginViewModel(IPackageSettings packageSettings, IPullRequestSessionManager sessionManager,
+            IWpfTextViewHost wpfTextViewHost, bool isDiffView)
+        {
+            this.packageSettings = packageSettings;
+            this.sessionManager = sessionManager;
+            this.wpfTextViewHost = wpfTextViewHost;
+            this.isDiffView = isDiffView;
+
+            Visible = IsVisible();
+            packageSettings.PropertyChanged += PackageSettings_PropertyChanged;
+        }
+
+        public void Dispose()
+        {
+            packageSettings.PropertyChanged -= PackageSettings_PropertyChanged;
```

Comment:
This should follow the established Dispose pattern that we use everywhere else in the codebase